packages/react-native-reanimated/src/mutables.ts

```diff
@@ -15,20 +15,22 @@ function shouldWarnAboutAccessDuringRender() {
   return __DEV__ && isReactRendering() && !isFirstReactRender();
 }
 
-function checkInvalidReadDuringRender() {
+function checkInvalidReadDuringRender(stack?: string) {
   if (shouldWarnAboutAccessDuringRender()) {
     logger.warn(
       "Reading from `value` during component render. Please ensure that you don't access the `value` property nor use `get` method of a shared value while React is rendering a component.",
-      { strict: true }
+      { strict: true },
+      { stack }
     );
   }
 }
 
-function checkInvalidWriteDuringRender() {
+function checkInvalidWriteDuringRender(stack?: string) {
   if (shouldWarnAboutAccessDuringRender()) {
     logger.warn(
-      "Writing to `value` during component render. Please ensure that you don't access the `value` property nor use `set` method of a shared value while React is rendering a component.",
-      { strict: true }
+      '!!Writing to `value` during component render. Please ensure that you do not access the `value` property or use `set` method of a shared value while React is rendering a component.',
+      { strict: true },
+      { stack }
```

Comment:
This change will add one more frame in the call stack, so I think that the Error object should be crated as close to the function exposed from reanimated as possible. I don't think that moving error object anywhere else is a valid solution. It will introduce more hassle and doesn't address the real problem. See my [comment](https://github.com/software-mansion/react-native-reanimated/pull/6788#issuecomment-2529181537) below explaining why.

Reply:
@piaskowyk from skimming hermes code I think the `.stack` is a getter that will format a string so that would be slow. I think creating the Error() itself might be cheap and then I could only pass the error reference itself and access `.stack` later.